modules/opengl/src/texture/samplerobject.cpp

```diff
@@ -0,0 +1,82 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2022 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/opengl/texture/samplerobject.h>
+
+namespace inviwo {
+
+SamplerObject::SamplerObject() : currentlyBoundTextureUnit_{0} { glGenSamplers(1, &id_); }
+
+SamplerObject::~SamplerObject() {
+    /*if (id_ != 0) {
+        glDeleteSamplers(1, &id_);
+    }*/ // Causes "unable to upload renderui.vert"
+}
+
+// SamplerObject::SamplerObject(const Texture& tex) {
+//	SamplerObject();
+//	bind(tex.getID); }
+
+void SamplerObject::bind(GLuint textureUnitNumber) {
+    glBindSampler(textureUnitNumber, id_);
+    currentlyBoundTextureUnit_ = textureUnitNumber;
```

Comment:
Not needed as of right now, but I think it might be useful for debugging to figure out which unit you are working on. Mostly for debugging though. Thus it can be removed. Though, every time you bind, you need to specify which unit to bind to. If you have to bind lots of times, it might be nicer to have what unit it's bound to rather than setting it manually in every bind call.

Reply:
In out current approach we bind all texture unit every time we change shader anyway so I don't think that is a problem.. There is a limited number of texture units so you have to take care of tracking and reusing them